File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public ArrayList<Integer> solution(int[] arr, int[][] queries) {
        ArrayList<Integer> answer = new ArrayList<>();

        for (int i = 0; i < queries.length; i++){
            int s = queries[i][0], e = queries[i][1], k = queries[i][2];
            ArrayList<Integer> tmp = new ArrayList<>();
            for (int j = s; j <= e; j++){
                if ( arr[j] > k) tmp.add(arr[j]);
            }
            if (tmp.isEmpty()) {
                answer.add(-1);
            } else {
                answer.add(Collections.min(tmp));
            }
        }

        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    
    ans = []
    arrcopy = arr
    

    for i in range(len(queries)):
        s, e, k = queries[i][0], queries[i][1], queries[i][2]
        arrcopy = arr[s:e+1]
        arrcopy.sort()
        check = 0
        
        for j in range(len(arrcopy)):
            if arrcopy[j] > k:
                ans.append(arrcopy[j])
                check = 1
                break
        
        if not check:
            ans.append(-1)
                
    return ans